code-challenges.js

```diff
@@ -9,60 +9,186 @@
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
-var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
+// var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
+// var catsOnly = []
+
+// const findCats = (array) => {
+//     catsOnly= array.filter(value => {
+//         return value.animal == 'cat'
+//     })
+//     return catsOnly
+// }
+
+// console.log(findCats(toonimals));
+
+
+//DISREGARD CODE BELOW. MY ANSWER IS ABOVE
+//However if any of you had some time why is the solution I am attempting not working in the code below. I would love to know. Thank You!
+//***************************************************************************************************************************************************************************************************** */
+// var catsOnly = [{name: "", animal: ""}]
+// var catName = ''
+// var catFound = ''
+// // var testCats = [ { name: "hello" } ]
+// // testCats[0].name = 'hi'
+
+
+// const findCats = (array) => {
+//     for (let index = 0; index < array.length; index++) {
+//         if(toonimals[index].animal == "cat"){
+//             // console.log(toonimals[index].name, toonimals[index].animal);
+
+//             catName= toonimals[index].name 
+//             catFound = toonimals[index].animal 
+//             console.log(catName);
+//             console.log(catFound);
+
+//             catsOnly[index].name = catName
+//             catsOnly[index].animal = catFound
+
+
+//             // catsOnly[index] = toonimals[index].name 
+//             // catsOnly[index] = toonimals[index].animal 
+//             // console.log(catsOnly)
+//         }
+//         // return catsOnly.join("")
+//     }
+// }
+
+
+// console.log(findCats(toonimals));
+// // console.log(toonimals[0].name);
+// // console.log(testCats[0].name);
 
 
 
 
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+
+// var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
+// var catsOnly = []
+
+// const findCats = (array) => {
+//     catsOnly= array.filter(value => {
+//         return value.animal !== 'cat'
+//     })
+//     for (let index = 0; index < catsOnly.length; index++) {
+//         console.log(catsOnly[index].name);
+//     }
+// }
```

Comment:
this totally works! but we don't want a console.log to replace our final return in the function. my question would be why you use a filter to then use a loop. could we use a filter and then a map while returning that map? this would allow us a final return instead of console.logs which are just there for testing and debugging.